tools/filter_cells_and_genes/filter_cells_and_genes.xml

```diff
@@ -0,0 +1,270 @@
+         <output name="output_matrix" file="counts_with_mito_scater_low.abundances_filtered.tab" ftype="tabular"/>
+    </test>
+    <test>
+         <param name="file" value="counts_with_mito.tab" ftype="tabular"/>
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="TRUE"/>
+         <param name="header_mito" value="MT-"/>
+         <param name="package" value="scater"/>
+         <param name="gene_filter" value="min.cells"/>
+         <param name="min_cells" value="3"/>
+         <output name="output_plot" file="counts_with_mito_scater_min.cells_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_with_mito_scater_min.cells_filtered.tab" ftype="tabular"/>
+    </test>
+    <test>
+         <param name="file" value="counts_with_mito.tab" ftype="tabular"/>
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="TRUE"/>
+         <param name="header_mito" value="MT-"/>
+         <param name="package" value="Seurat"/>
+         <param name="min_genes" value="10"/>
+         <param name="min_cells" value="3"/>
+         <output name="output_plot" file="counts_with_mito_Seurat_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_with_mito_Seurat_filtered.tab" ftype="tabular"/>
+    </test> 
+   </tests>
+   <help>
+
+**What it does**
+
+Filter_cells_and_genes takes a tabular table of raw read counts, each column corresponding to a cell and each row corresponding to a
+gene, and returns (1) a PDF with quality controls plots (like the distribution of library sizes or expressed genes by cell, frequency of 
+expression or the most expressed genes) and (2) a filtered table of counts.
+
+You can filter out low quality cells from your raw expression table :
+
+- Based on expression level of mitochondrial genes :
+
+    - Using R package scater: remove cells that detect mitochondrial genes 3 times higher than the median of all cells
+    - Using R package Seurat: if cells expressed more than 20% of mitochondrial genes, they are removed.
+
+- Using R package scater : search cells with low library size and few detected genes based on the deviation from the median of all cel$
+
+- Using R package Seurat : Remove all cells that detected at least n genes.
+
+Also you can filter out low quality genes from your raw expression table :
+
+- Using R package scater :
+
+    - Remove all genes that aren't detected in at least n cells
+    - Filter out genes based on its average count (which is adjust for library size between cells). It is more stringent.
+
+- Using R package Seurat : Remove all genes that aren't detected in at least n cells.
+
+
+**Inputs**
+
+The following table gives an example of the tabular input that has to be provided to the tool:
```

Comment:
Note that the first line show cell identifiers and table report the expression of xx gene (aka xx lines).

(be nice with your users)

Reply:
ping @bellenger-l